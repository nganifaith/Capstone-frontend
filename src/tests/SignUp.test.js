import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createStore } from 'redux';
import rootReducer from '../reducers';

import SignUpPage from '../pages/SignUp';

const history = createMemoryHistory();
const store = createStore(rootReducer);

test('full app rendering/navigating', () => {
  const signUp = renderer.create(
    <Provider store={store}>
      <Router history={history}>
        <SignUpPage />
      </Router>
      ,
    </Provider>,
  ).toJSON();
  expect(signUp).toMatchSnapshot();
});
