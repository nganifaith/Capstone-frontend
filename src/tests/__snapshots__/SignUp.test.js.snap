// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full app rendering/navigating 1`] = `
Array [
  <div
    className="sc-bdfBQB kSOTPF"
  >
    <div
      className="sc-gsTEea ccZMNe"
    >
      <form
        className="sc-dlfnuX iXEfZY"
        onSubmit={[Function]}
      >
        <div
          className="Error"
        >
          
        </div>
        <label
          htmlFor="name"
        >
          Name
          <input
            id="name"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </label>
        <label
          htmlFor="email"
        >
          Email
          <input
            id="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </label>
        <label
          htmlFor="password"
        >
          Password
          <input
            id="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </label>
        <label
          htmlFor="confirm"
        >
          Confirm Password
          <input
            id="confirm"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </label>
        <button
          className="sc-hKgJUU hpcpuN"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  ",",
]
`;
